Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverLocation.hs

```diff
@@ -151,10 +165,13 @@ findAllDriverLocations driverIds mbDriverPositionInfoExpiry now = do
   catMaybes <$> mapM fromTType' (fromRight [] geoms)
 
 getDriverLocations ::
-  MonadFlow m =>
+  (MonadFlow m, MonadReader r m, HasField "enableLocationTrackingService" r Bool) =>
   UTCTime ->
   m [DriverLocation]
 getDriverLocations before = do
+  enableLocationTrackingService <- asks (.enableLocationTrackingService)
+  when enableLocationTrackingService $ do
+    throwError InvalidLocationTrackingException
```

Comment:
Why are we throwing error in these cases?

Reply:
we are throwing this errors in queries and on base level functions because by this we can know that when location tracking service env is enabled and some Query is called on Driver Location table from any API. As ideally Driver location table should not be queried upon if Location Service is enabled. So, @NikithShetty suggested to add this checks on base level to know if any error comes that means we missed handling that API.